pandas/_libs/index.pyx

```diff
@@ -803,7 +803,7 @@ cdef class BaseMultiIndexCodesEngine:
         int_keys : 1-dimensional array of dtype uint64 or object
             Integers representing one combination each
         """
-        level_codes = list(target._recode_for_new_levels(self.levels))
+        level_codes = list(target._recode_for_new_levels(self.levels, copy=True))
```

Comment:
Is a copy needed if you’re about to cast to list?

Reply:
`_recode_for_new_levels` is a generator, so I think the `list` is just collecting the result. But I think the `copy=True` is still needed because the codes are being modified in the for loop below?

Reply:
Makes sense thanks